### ISSUE TITLE
How should we handle Patients with multiple Normal samples?

### ISSUE THREAD

**Issue by shuntsman-ucsf** [UCSF-Costello-Lab/LG3_Pipeline#93]
I have a patient with 2 normal fastq.  The recal step fails due to multiple files listed as "Normal" in the patient_ID_conversions table
```
- SAMPLES=SC299898_CGCTGATC_L002 SC299898_CGCTGATC_L006 SC299982_AACAACCA_L001 SC299982_AACAACCA_L002
- NORMAL=SC299898_CGCTGATC_L002
SC299898_CGCTGATC_L006
...
...
grep: the -P option only supports a single pattern
ERROR: NORMAL 'SC299898_CGCTGATC_L002
SC299898_CGCTGATC_L006' is not part of SAMPLES: SC299898_CGCTGATC_L002 SC299898_CGCTGATC_L006 SC299982_AACAACCA_L001 SC299982_AACAACCA_L002
```

What is the correct way to handle it?
- relabel the 2nd normal as "Normal2" or something like that, then re-run through recal (how does this affect downstream?)
- merge Normal1 and Normal2 prior to recal
- discard the 2nd recal fastq from analysis

(not urgent since I have plenty of others to recal, but will want to run eventually)

**Comment by ivan108** [comment]
Yes, you need to go through merging procedure, only 1 normal is allowed...

**Comment by shuntsman-ucsf** [comment]
OK, so correct procedure for multiple Normals is:

- Trim
- Align
- Merge Normals, adjust patientTable
- Recal
- Merge Tumors, adjust patientTable
- Recal_Pass2
- Pindel, Mutect
- PostMutect

**Comment by ivan108** [comment]
Not quite:
1. Trim
2. Align
3. Recal
4. Merge all groups, Normals and/or Tumors
5. Change patient table, replace duplicated sampleIDs with merged IDs
6. Recal_pass2
7. Pindel,Mutect
8. PostMutect

**Comment by shuntsman-ucsf** [comment]
The recal step only recognizes 1 Normal from the patient_ID_conversions.tsv or it crashes.

example output from run:
```
......
- SAMPLES=SC299979_ACAGCAGA_L001 SC299979_ACAGCAGA_L005 SC300108_CCTCTATC_L001 SC300108_CCTCTATC_L002
- NORMAL=SC299979_ACAGCAGA_L001
SC299979_ACAGCAGA_L005
- ILIST=/home/shared/cbc/software_cbc/LG3_Pipeline-2018-10-17-patched/resources/SeqCap_EZ_Exome_v3_capture.interval_list
- EMAIL=scott.huntsman@ucsf.edu
grep: the -P option only supports a single pattern
ERROR: NORMAL 'SC299979_ACAGCAGA_L001
SC299979_ACAGCAGA_L005' is not part of SAMPLES: SC299979_ACAGCAGA_L001 SC299979_ACAGCAGA_L005 SC300108_CCTCTATC_L001 SC300108_CCTCTATC_L002
```


- How should additional "Normal"s  be labeled in the conversions.tsv prior to merging? 
- Does the label (whether Normal or Primary, Recurrence1, etc...) affect how it is handled in the recal step?

**Comment by ivan108** [comment]
Yes, you are right, the LG3 pipeline was designed to work with exactly one Normal per patient.

The work around involves merging Normals, and it requires to change patient "conversion" file (PCF).
Here are step by step directions:
- On the first pass your PCF1 should contain all normal samples and all other samples, but only one normal is marked as "Normal", the other normal(s) can be labeled anything but "Normal", e.g. "Primary" or "Recurrence1". At this step those labels are not used for anything but identifying a single normal sample. 
- after PCF1 is setup, you run Trim, Align and Recal as usual.
- next run Merge (or Merge_QC) on normal samples, and give the merged sample a new ID.
To run Merge you need to provide variable SAMPLES and SAMPLE, in addition to the usual variables, e.g.:
SAMPLES="SC299898_CGCTGATC_L002 SC299898_CGCTGATC_L006"
SAMPLE=SC299898_normal_merged
- next update PCF1, by replacing all normal samples of the patient with a single normal merged sample, labeled as "Normal" (=PCF2).
- next run Recal_pass2
- run Pindel and MutDet as usual
- run PostMut as usual

Hope it makes sense...

**Comment by HenrikBengtsson** [comment]
@shuntsman-ucsf, feel free to propose improvements/how you think this can be best handled.  You're currently the only one that is using the pipeline this way, so you're the one best positioned to provide such suggestions.
